services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -88,6 +90,28 @@ class DynamicSidecarsScheduler:  # pylint: disable=too-many-instance-attributes
     _trigger_observation_queue: Queue = field(default_factory=Queue)
     _observation_counter: int = 0
 
+    def toggle_observation_cycle(self, node_uuid: NodeID, disable: bool) -> bool:
+        """
+        returns True if it manged to toggle the current state
```

Comment:
THOUGHT: this is an excellent example for a defect that I see a lot!

In principle this function is designed to use the return to inform whether it succeeded or failed. 
Nonetheless under certain conditions on ``node_uuid``  it can raise an exception! ... which it is as well a failure . (not to mention the other )

IMO this is creates ambiguity in what to expect on the behavior of the function . In Robust Python (the book on my desk) they discuss in detail this problem and suggest a design w/o exceptions raised but rather a type for results and errors  ... 
That would lead to ... 
```python

def toggle_observation_cycle(self, node_uuid: NodeID, disable: bool) -> Union[Result, Error]:
     ...
     try:
        return result
     except MyException as err:
        return Error(err) 
```